streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -362,30 +384,13 @@ public void recordThroughput(final Sensor sensor, final long value) {
         sensor.record(value);
     }
 
-    public final Map<String, String> tagMap(final String... tags) {
-        final Map<String, String> tagMap = new LinkedHashMap<>();
-        tagMap.put("client-id", threadName);
-        if (tags != null) {
-            if ((tags.length % 2) != 0) {
-                throw new IllegalArgumentException("Tags needs to be specified in key-value pairs");
-            }
-
-            for (int i = 0; i < tags.length; i += 2) {
-                tagMap.put(tags[i], tags[i + 1]);
-            }
-        }
-        return tagMap;
-    }
-
-
     private Map<String, String> constructTags(final String scopeName, final String entityName, final String... tags) {
         final String[] updatedTags = Arrays.copyOf(tags, tags.length + 2);
         updatedTags[tags.length] = scopeName + "-id";
         updatedTags[tags.length + 1] = entityName;
-        return tagMap(updatedTags);
+        return threadLevelTagMap(updatedTags);
```

Comment:
This makes me thinking: should we clarify in our javadoc that user customized metrics would be recorded at the thread-level specifically, with additional scope tag?

Reply:
That would be good idea. Do you want to update the javadocs in KIP-444?

Reply:
SG.

Reply:
Thank you!